llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -1552,7 +1552,7 @@ RISCVTTIImpl::getArithmeticReductionCost(unsigned Opcode, VectorType *Ty,
   }
 
   // IR Reduction is composed by two vmv and one rvv reduction instruction.
-  if (TTI::requiresOrderedReduction(FMF)) {
+  if (TTI::requiresOrderedReduction(FMF) && ElementTy->isFloatingPointTy()) {
```

Comment:
I think a cleaner way to express this change is to simply sink the check for requiresOrderedReduction under the FADD case in the switch below.  That's the only supported reduction kind with an ordered variant.